code/game/machinery/vendors/custom_vendors.dm

```diff
@@ -0,0 +1,86 @@
+/obj/machinery/economy/vending/custom
+	name = "\improper CrewVend 3000"
+	refill_canister = null
+	always_deconstruct = TRUE
+	var/obj/item/eftpos/linked_pos
+
+/obj/machinery/economy/vending/custom/Destroy()
+	if(!isnull(linked_pos))
+		linked_pos.linked_vendors -= src
+		linked_pos = null
+	return ..()
+
+/obj/machinery/economy/vending/custom/locked()
+	return isnull(linked_pos) || linked_pos.transaction_locked
+
+/obj/machinery/economy/vending/custom/get_vendor_account()
+	return linked_pos?.linked_account || ..()
+
+/obj/machinery/economy/vending/custom/item_interaction(mob/living/user, obj/item/used, list/modifiers)
+	if(istype(used, /obj/item/eftpos))
+		visible_message("[src] beeps as [user] links it to [used].", "You hear something beep.")
+		if(!isnull(linked_pos))
+			linked_pos.linked_vendors -= src
+		linked_pos = used
+		var/obj/item/eftpos/pos = used
+		pos.linked_vendors += src
+		return ITEM_INTERACT_COMPLETE
+	else if(isnull(linked_pos))
+		to_chat(user, "<span class='warning'>You need to link a point of sale device first!</span>")
+		return ITEM_INTERACT_COMPLETE
+	else if(locked())
+		return ..()
+	else if(used.tool_behaviour == TOOL_CROWBAR)
+		return ..()
```

Comment:
was crowbar_act not being called properly before the item_interaction proc?

Reply:
It wasn't working without that at one point, but seems to do OK now (once I fixed up crowbar_act by making sure custom vendors alway have component_parts).